JOSS paper

PhilBoileau opened the pull request:
An initial draft of the JOSS paper.

nhejazi commented on paper/paper.md:
```diff
@@ -26,18 +29,60 @@ affiliations:
     index: 3
   - name: Division of Epidemiology and Biostatistics, School of Public Health, University of California, Berkeley
```
```suggestion
  - name: Division of Biostatistics, School of Public Health, University of California, Berkeley
```
I think the Biostatistics and Epidemiology Divisions have become independent after the merger a year or two ago.

nhejazi commented on paper/paper.md:
```diff
@@ -26,18 +29,60 @@ affiliations:
+date: 03 February 2021
 bibliography: paper.bib
 ---
 
```
High-level, the contents of the paper look good. Some general changes would be
* naming + citing exactly the covariance matrix estimators that are already implemented (stating the version number of the package for clarity)
* discussing more the visualization functionality, with a simple example. for example, we could grab the `README` example, mention the estimators that we select over, and then display one of a table and the nice plots that shows the performance within estimator families.
* in my most recent experience with submitting to JOSS, the editor asked that we re-organize the paper into short sections (as in https://github.com/nhejazi/txshift/blob/master/paper/paper.md). i think this is a good way to go, so that we have short paragraphs describing what `cvCovEst` does, what the problem is, and how we solve the problem (with technical details relegated to the methods paper)

PhilBoileau replied:
Done! See most recent comments.

PhilBoileau commented:
I've made a number of changes to reflect @nhejazi's suggestions and comments:

+ Organized existing paragraphs under descriptive headers
+ Added a Summary section that gives a high-level overview of the package
+ Added a table of implemented estimators that includes a brief description and citation
+ Added a Toy Dataset Example section to showcase a typical application of `cvCovEst()`. Includes a call to `plot()` and the resulting summary graph.

The draft is ready for a second round of review.

nhejazi reviewed (approved):
added some updates to the paper, now LGTM

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+When the observations in a dataset far outnumber the features, asymptotic theory
+and empirical evidence have demonstrated the sample covariance matrix to be the
+optimal estimator of this parameter. This assertion does not hold when the
+number of observations is commensurate or smaller than the number of features.
```
```suggestion
number of observations is commensurate with or smaller than the number of features.
```

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+the many possible candidates remains a largely unaddressed challenge. The
+`cvCovEst` package addresses this methodological gap through its implementation
+of a cross-validated framework for covariance matrix estimator selection. This
+data-adaptive procedure's selections are asymptotically optimal under minimal
+assumptions, meaning they are equivalent to the selections one would make if the
+true data-generating processes were known [@vdl2003unified].
```
```suggestion
data-adaptive procedure's selections are asymptotically optimal under minimal
assumptions -- in fact, they are equivalent to the selections that would be made if given access to full knowledge of the true data-generating processes (i.e., an oracle selector) [@vdl2003unified].
```
The initial wording felt awkward since the second clause following "meaning" was not directly implied by the preceding clause.

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+
+# `cvCovEst` Framework
+
+The solution provided by `cvCovEst` a general, cross-validation-based,
```
```suggestion
The solution provided by `cvCovEst` is a general, cross-validation-based,
```

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+@vaart2006 on data-adaptive estimator selection to high-dimensional covariance
+matrix estimation [@boileau2021]. Here, optimality is defined as choosing an
+estimator with an equivalent risk difference to that which would have been
+selected were the underlying data-generating distribution _known_.
```
```suggestion
selected were the underlying data-generating distribution _completely known_.
```

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+computation. Parallelization relies on the suite of `future` packages
+[@future] by way of the `origami` package [@origami].
+
+Table 1: Covariance matrix estimators implemented as of version 0.3.4.
```
```suggestion
Table 1: Covariance matrix estimators implemented as of [version 0.3.4](https://github.com/PhilBoileau/cvCovEst).
```
We should eventually replace the link to the package with one specifying a git tag / GitHub release of v0.3.4

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+|Estimator | Implementation | Description |
+|----------|----------|-------------|
+| Sample covariance matrix | `sampleCovEst()` | The sample covariance matrix. |
+| Hard thresholding [@Bickel2008_thresh] | `thresholdingEst()` | Applies a hard thresholding operator to the entries of the sample covariance matrix |
```
```suggestion
| Hard thresholding [@Bickel2008_thresh] | `thresholdingEst()` | Applies a hard thresholding operator to the entries of the sample covariance matrix. |
```
Insert "."

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+| Sample covariance matrix | `sampleCovEst()` | The sample covariance matrix. |
+| Hard thresholding [@Bickel2008_thresh] | `thresholdingEst()` | Applies a hard thresholding operator to the entries of the sample covariance matrix |
+| SCAD thresholding [@rothman2009;@fan2001] | `scadEst()` | Applies the SCAD thresholding operator to the entries of the sample covariance matrix.|
+| Adaptive LASSO [@rothman2009] | `adaptiveLassoEst()` | Applies the adaptive LASSO thresholding operator to the entries of the sample covariance matrix |
```
```suggestion
| Adaptive LASSO [@rothman2009] | `adaptiveLassoEst()` | Applies the adaptive LASSO thresholding operator to the entries of the sample covariance matrix. |
```
Insert "."

nhejazi commented on paper/paper.md:
```diff
@@ -1,43 +1,204 @@
+[Comprehensive `R` Archive Network](https://CRAN.R-project.org/package=cvCovEst).
+Its development version can be found on
+[GitHub](https://github.com/PhilBoileau/cvCovEst). Documentation and examples
+are contained in each version's manual pages and vignette.
```
```suggestion
are contained in each version's manual pages, vignette, and `pkgdown` website (at https://philboileau.github.io/cvCovEst).
```

nhejazi reviewed (approved):
The JOSS paper looks to be in very good shape. I have only minor comments and/or suggestions, but these should not hold up an initial submission.

PhilBoileau commented:
> The JOSS paper looks to be in very good shape. I have only minor comments and/or suggestions, but these should not hold up an initial submission.

Thank you for these edits, they look great. The fixing of typos and punctuation is also much appreciated. If Sandrine is on board, we should be able to submit this by the end of the week.
